tests/unit/s2n_ecc_preferences_test.c

```diff
@@ -18,6 +18,21 @@
 #include "s2n_test.h"
 #include "tls/s2n_tls_parameters.h"
 
+static const struct s2n_ecc_preferences *ALL_ECC_PREFERENCES[] = {
+    &s2n_ecc_preferences_20140601,
+    &s2n_ecc_preferences_20200310,
+    &s2n_ecc_preferences_20201021,
+    &s2n_ecc_preferences_20210816,
+    &s2n_ecc_preferences_20230623,
+    &s2n_ecc_preferences_20240501,
+    &s2n_ecc_preferences_20240603,
+    &s2n_ecc_preferences_default_fips,
+    &s2n_ecc_preferences_test_all,
+    &s2n_ecc_preferences_null,
+};
+
+static const size_t ALL_ECC_PREFERENCES_COUNT = 10;
```

Comment:
This will require developers to manually add all future ecc preferences to your test.

How we've done this for other components of the security policy is to iterate over all security policies: 
* https://github.com/aws/s2n-tls/blob/a7862238a9d6703e0a3e2d7ea4341de8d57429bd/tests/unit/s2n_signature_scheme_test.c#L29
* https://github.com/aws/s2n-tls/blob/a7862238a9d6703e0a3e2d7ea4341de8d57429bd/tests/unit/s2n_cipher_suites_test.c#L49

All security policies -> all reachable ecc preferences. You'll recheck some, but that's cheap.